doespy/doespy/etl/steps/colcross/subplots/bar_hooks.py

```diff
@@ -0,0 +1,118 @@
+from doespy.etl.steps.colcross.components import AxisConfig, Metric, axis_formatter
+from doespy.etl.steps.colcross.hooks import default_hooks
+import gossip
+import typing
+from typing import Dict, List, Optional, Tuple
+
+import numpy as np
+import pandas as pd
+from enum import Enum
+
+from matplotlib import pyplot as plt
+
+
+class BarHooks(str, Enum):
+
+    SubplotPre = "BarSubplotPre"
+
+    ArtistPre = "BarArtistPre"
+
+    ArtistPost = "BarArtistPost"
+
+    SubplotPost = "BarSubplotPost"
+
+
+##################################
+
+
+# TODO [nku] for each can write a docstring to explain the arguments
+@default_hooks.register(BarHooks.SubplotPre)
+def demo(
```

Comment:
are there any examples somewhere that show what we could do?

Reply:
In the custom etl steps of the demo project, we have an example that shows how you can add custom code to the locations where we define these hooks:

https://github.com/nicolas-kuechler/doe-suite/blob/hosts/demo_project/doe-suite-config/does_etl_custom/etl/plots/colcross.py

In this example, the goal is to add a watermark label on each subplot [Resulting Bar Plot with Watermark](https://github.com/nicolas-kuechler/doe-suite/blob/0844365e7412b6bd63ae0dae18aa05e8af2acd13/demo_project/doe-suite-results-super/demo02-colcross/custom/%24metrics%24%3Dtime.pdf).


- we extend the `SubplotConfig` with a `watermark` attribute (so that we can define the content of the watermark from the `yaml` config of the step:
  ```
  class MyCustomSubplotConfig(SubplotConfig):
      # INFO: We extend the default config and add a new attribute
      watermark: str = None
  ```
- we create a new loader, `MyCustomColumnCrossPlotLoader,` which replaces the default `SublotConfig` with the custom config containing the watermark.
  ```
  class MyCustomColumnCrossPlotLoader(BaseColumnCrossPlotLoader):
      # INFO: We provide a custom subplot config that extends the default config
      #       and override it here
      cum_subplot_config: List[MyCustomSubplotConfig]
  ```
- we register a new function that adds the watermark to the subplot. We register it in the `CcpHooks.SubplotPostChart` hook, which is called after the chart has been created in the subplot. 
  ```
  @MyCustomColumnCrossPlotLoader.blueprint().register(CcpHooks.SubplotPostChart)
  def apply_watermark(...)
  ```